main.py

```diff
@@ -107,7 +114,25 @@ def setFont(self, widget, textBuffer):
             tag = textBuffer.create_tag(tagName, font_desc=fontDescriptor)
         textBuffer.apply_tag(tag, *textBuffer.get_selection_bounds())
         fontChooserDialog.destroy()
+        
+    def openBuffer(self, widget, textBuffer):
+        openDialog = Gtk.FileChooserDialog()
+        openDialog.set_action(Gtk.FileChooserAction.OPEN)
+        openDialog.add_button("Cancel", Gtk.ResponseType.CANCEL)
+        openDialog.add_button("OPEN", Gtk.ResponseType.OK)
+
+        response = openDialog.run()
+        if response != Gtk.ResponseType.OK:
+            openDialog.destroy()
+            return
+        path = os.path.join(openDialog.get_current_folder(), openDialog.get_filename())
+        bounds = textBuffer.get_bounds()
+        deserializationFormat = textBuffer.register_deserialize_tagset()
+        deserial = textBuffer.deserialize(textBuffer, deserializationFormat, textiter, data)
```

Comment:
You are currently using `data` in this invocation, but this is not defined anywhere. Data should be the bytes you read from the file. So what you should do is put this line inside the `with open(...` block, read the file data into a data variable `data = inputFile.read()`, then execute the deserialization method to get the text. Pass that text into the `set_text(text)` method of the text buffe, and you should be done.